src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -10972,8 +10976,10 @@ public abstract partial class TextReader : System.MarshalByRefObject, System.IDi
         public virtual System.Threading.Tasks.ValueTask<int> ReadBlockAsync(System.Memory<char> buffer, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual string? ReadLine() { throw null; }
         public virtual System.Threading.Tasks.Task<string?> ReadLineAsync() { throw null; }
+        public virtual System.Threading.Tasks.ValueTask<string?> ReadLineAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
```

Comment:
That is another non-obvious thing about contributing to this repo, but the `new-api-needs-documentation` label applied by our bot requires that every new public method needs to have XML documentation (the comments that start with ///). These comments are later parsed by another bot and fed to https://github.com/dotnet/dotnet-api-docs/ and finally shown at https://docs.microsoft.com/ and also included in IntelliSense package for VS.

Please add the full XML docs for each new public method (to the `.cs` files located in `/src/`, not in `/ref/`). To do that, you can just copy the comments of existing methods that don't accept `CancellationToken` from [official docs](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.readlineasync?view=net-6.0) and extend them with a comment for `CancellationToken`:

```xml
<param name="cancellationToken">The token to monitor for cancellation requests.</param>
```

It's also a great place to mention what are the guarantees after cancellation takes place.

Reply:
I assumed you meant `src/libraries/System.Private.CoreLib/src/System/IO/TextReader.cs`, etc. and added XML documentation comments there. (There are no existing comments in those files, but I assume that's because this is a new process for new methods only; let me know if that's not correct.)